wordpress-mobile/gutenberg-mobile
# Add media edit icon to image block

Fixes #1583

`Gutenberg` PR -> https://github.com/WordPress/gutenberg/pull/19723

`WPiOS` related PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/13291
`WPAndroid` related PR: https://github.com/wordpress-mobile/WordPress-Android/pull/11175

<img src="https://user-images.githubusercontent.com/7040243/72620911-fd517e80-391e-11ea-802b-e8968d2a37e6.png" width="300">

This PR adds a button over Image blocks that shows two options:

1. Edit: shows the Media Editor so the user can edit the image
2. Replace: to replace the media

As for now, the button is enabled to show only in iOS, since the feature is in the first stages of development in Android.

### To test

In order for this PR to work you'll have to checkout my fork-branch (I don't have permissions on the main `gutenberg` repo). You can [use a tool like hub to make that easier](https://hub.github.com/). Eg.: `hub pr checkout 19723`

Then, for both Android and iOS (to enable in Android just change `showMediaEditorButton` to `true`:

1. Check that the button appears only for Images
2. Check that the button appears only for images that are uploaded (and not being uploaded)
3. Tap the button and checks that a Picker appears

PR submission checklist:

- [ ] I have considered adding unit tests where possible. — Refer to Gutenberg's parent PR. I had some issues using `shallow`.
